src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -3504,6 +3507,17 @@ public bool ShouldReleaseTransactionBecauseFlushIsWaiting(IndexingStatsScope sta
                 }
             }
 
+            if (Configuration.ManagedAllocationsBatchLimit != null)
+            {
+                var currentManagedAllocations = new Size(GC.GetAllocatedBytesForCurrentThread(), SizeUnit.Bytes);
```

Comment:
How expensive is: `GC.GetAllocatedBytesForCurrentThread()`?
Don't we have this number from here:
https://github.com/ravendb/ravendb/blob/3ad961e29b41eda13444a9d5d34c4491966a74c6/src/Raven.Server/Documents/Indexes/Index.cs#L3821

Reply:
This is pretty light weighted and was requested by oren few years ago ;)

It is not about the memory, it is about the number of objects.

Reply:
Can we do it every 100 or 1000 items? Not on every document?